src/player/player-engine.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2023 zheng qian. All Rights Reserved.
+ *
+ * @author zheng qian <xqq@xqq.im>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type MediaInfo from "../core/media-info";
+
+export default interface PlayerEngine {
+    destroy(): void;
+    on(event: string, listener: (...args: any[]) => void): void;
+    off(event: string, listener: (...args: any[]) => void): void;
+    attachMediaElement(mediaElement: HTMLMediaElement): void;
+    detachMediaElement(): void;
+    load(): void;
+    unload(): void;
+    play(): Promise<void>;
+    pause(): void;
+    seek(seconds: number): void;
+    readonly mediaInfo: MediaInfo | undefined;
+    readonly statisticsInfo: any | undefined;
+}
```

Comment:
I would like to PlayerEngine to abstract class and lift up non main/dedicated related code to abstract class.